File: Practice/18/Python/eurotour.py
dictionary = str('hallo, klempner, das, ist, fantastisch, fluggegecheimen')

def remover(d: str) -> str:
    d1 = d.replace(',', '')
    dmerged = d1.replace(' ', '')
    return dmerged

def counter(d:str, letter) -> float:
    q = int(0)
    for i in range(len(d)):
        if (d[i] == letter):
            q = q + 1
    return q

dmerged = remover(dictionary)
prob = float(1)
stop = input('Введите стоп-слово\n')
for i in range(len(stop)):
    prob = prob * (counter(dmerged, stop[i]) / len(dmerged))
print('Вероятность для стоп-слова:', prob)

File: Practice/18/C++/eurotour/eurotour.cpp
#include <iostream>
using namespace std;

string dictionary = "hallo, klempner, das, ist, fantastisch, fluggegecheimen";

string remover(string d)
{
    for (int i = 0; i < d.length(); i++)
        if (d[i] == ',' or d[i] == ' ')
        {
            d.erase(i, 1);
            i--;
        }
    return d;
}

double counter(string dmerged, char letter)
{
    int q = 0;
    for (int i = 0; i < dmerged.length(); i++)
    {
        if (dmerged[i] == letter)
            q++;
    }
    return q;
}
int main()
{
    setlocale(LC_ALL, "Russian");

    double prob = 1;
    string dmerged, stop;
    dmerged = remover(dictionary);
    cout << "Введите стоп-слово\n";
    cin >> stop;
    for (int i = 0; i < stop.length(); i++)
        prob = prob * (counter(dmerged, stop[i]) / dmerged.length());
    cout << "Вероятность для стоп-слова: " << prob;
}